burstphoto/align.metal

```diff
@@ -156,12 +326,15 @@ kernel void compute_tile_differences(texture2d<float, access::read> ref_texture
             // compute the indices of the pixels to compare
             int ref_tile_x = x0 + dx1;
             int ref_tile_y = y0 + dy1;
-            int comp_tile_x = x0 + dx0 + dx1;
-            int comp_tile_y = y0 + dy0 + dy1;
+            int comp_tile_x = ref_tile_x + dx0;
+            int comp_tile_y = ref_tile_y + dy0;
+            
             // if the comparison pixels are outside of the frame, attach a high loss to them
             if ((comp_tile_x < 0) || (comp_tile_y < 0) || (comp_tile_x >= texture_width) || (comp_tile_y >= texture_height)) {
```

Comment:
I think it should be possible to rewrite the loops so that they're only iterating over positions that are inside the frame. The diff can be updated to account for the ones outside of the frame outside of the loop.

This *may* speed things up since you can get rid of the branch inside your loop.

Reply:
This is not an easy task as we are iterating over the tiles in the reference frame and have to take into account the alignment vectors before we can decide if we are outside the frame or not. While I agree that it may be possible, I do not directly have an idea how to implement it.

Reply:
No reasonable solution found.